lib/processPbrMetallicRoughness.js

```diff
@@ -0,0 +1,774 @@
+/*global define*/
+define([
+    './addToArray',
+    './ForEach',
+    './numberOfComponentsForType',
+    './techniqueParameterForSemantic',
+    '../../Core/clone',
+    '../../Core/defined',
+    '../../Core/defaultValue',
+    '../../Core/WebGLConstants'
+], function(
+    addToArray,
+    ForEach,
+    numberOfComponentsForType,
+    techniqueParameterForSemantic,
+    clone,
+    defined,
+    defaultValue,
+    WebGLConstants) {
+    'use strict';
+
+    function webGLConstantToGlslType(webGLValue) {
+        switch (webGLValue) {
+            case WebGLConstants.FLOAT:
+                return 'float';
+            case WebGLConstants.FLOAT_VEC2:
+                return 'vec2';
+            case WebGLConstants.FLOAT_VEC3:
+                return 'vec3';
+            case WebGLConstants.FLOAT_VEC4:
+                return 'vec4';
+            case WebGLConstants.FLOAT_MAT2:
+                return 'mat2';
+            case WebGLConstants.FLOAT_MAT3:
+                return 'mat3';
+            case WebGLConstants.FLOAT_MAT4:
+                return 'mat4';
+            case WebGLConstants.SAMPLER_2D:
+                return 'sampler2D';
+            case WebGLConstants.BOOL:
+                return 'bool';
+        }
+    }
+
+    function glslTypeToWebGLConstant(glslType) {
+        switch (glslType) {
+            case 'float':
+                return WebGLConstants.FLOAT;
+            case 'vec2':
+                return WebGLConstants.FLOAT_VEC2;
+            case 'vec3':
+                return WebGLConstants.FLOAT_VEC3;
+            case 'vec4':
+                return WebGLConstants.FLOAT_VEC4;
+            case 'mat2':
+                return WebGLConstants.FLOAT_MAT2;
+            case 'mat3':
+                return WebGLConstants.FLOAT_MAT3;
+            case 'mat4':
+                return WebGLConstants.FLOAT_MAT4;
+            case 'sampler2D':
+                return WebGLConstants.SAMPLER_2D;
+        }
+    }
+
+    function generateTechnique(gltf, material, frameState, optimizeForCesium) {
+        var techniques = gltf.techniques;
+        var shaders = gltf.shaders;
+        var programs = gltf.programs;
+
+        var parameterValues = material.pbrMetallicRoughness;
+        for (var additional in material) {
+            if (material.hasOwnProperty(additional) && ((additional.indexOf('Texture') >= 0) || additional.indexOf('Factor') >= 0) || additional === 'doubleSided') {
+                parameterValues[additional] = material[additional];
+            }
+        }
+
+        var vertexShader = 'precision highp float;\n';
+        var fragmentShader = 'precision highp float;\n';
+
+        var skin = gltf.skins[0];
```

Comment:
It doesn't seem to say explicitly in the spec, but all of the test models at least have an empty `gltf.skins` . This could be the default behavior of exporters/coverters though, so I'll put a check just in case.